node/node.go

```diff
@@ -60,23 +60,28 @@ func New(cfg *Config) (StarkNetNode, error) {
 		}
 		cfg.DatabasePath = filepath.Join(dirPrefix, cfg.Network.String())
 	}
-
-	bc := blockchain.NewBlockchain()
-	return &Node{
-		cfg:        cfg,
-		blockchain: bc,
-		syncLoop:   sync.NewSyncLoop(bc, nil),
-	}, nil
+	return &Node{cfg: cfg}, nil
 }
 
 func (n *Node) Run() error {
 	log.Println("Running Juno with config: ", fmt.Sprintf("%+v", *n.cfg))
 
-	return n.syncLoop.Run()
+	var err error
+	n.db, err = db.NewDb(n.cfg.DatabasePath)
+	if err != nil {
+		return err
+	}
+	defer n.db.Close()
+	n.blockchain, err = blockchain.NewBlockchain(n.db, n.cfg.Network)
+	if err != nil {
+		return err
+	}
+	n.synchronizer = sync.NewSynchronizer(n.blockchain, gateway.NewGateway(n.cfg.Network))
```

Comment:
IMO, all instantiations should happen in `New` where we create the `Node`

Reply:
I did have all the instantiation in the New function but creating a db opens the db and we can't close it in the New function. This meant some of the tests were failing when we only wanted to check whether the config was set correctly or not.

I moved the db part to the Run function but then I had a mixture of instantiation across New and Run. So I decided that New will set the config while Run will handle all the instantiation and start the node.